src/pykeen/triples/triples_numeric_literals_factory.py

```diff
@@ -43,6 +45,24 @@ def create_matrix_of_literals(
     return num_literals, data_rel_to_id
 
 
+def resolve_function(function: Union[str, Callable], function_dict: Dict[str, Callable]) -> Callable:
+    if isinstance(function, str):
+        return function_dict[function]
+    elif isinstance(function, Callable):
+        return function
+    else:
+        raise ValueError(f"Invalid type of {function}: {type(function)}")
+
+
+_NUM_LITERALS_PREPROCESSING_FUNCTIONS = dict(
```

Comment:
we've already written utilities for resolving functions, see https://github.com/pykeen/pykeen/blob/7bb5dd6482234c4ccd933ffdbf8e2a6c559ba292/src/pykeen/nn/representation.py#L567-L569 for an example or more generally, check out https://github.com/cthoyt/class-resolver